### PR TITLE
Fix adding Dispose when private IDisposable class is present

### PR DESCRIPTION
**Pull request by kgorbenko**
Exercism fails to run tests over Error Handling exercise because it's test suite contains private class that implements IDisposable. In this case test runner adds IDisposable base type on test suite class, but instead of implementing Dispose method it finds it within private class and modifies it. As a result, test suite class is left without implementation of IDisposable and fails to compile.

Current solution takes into consideration only implementations of Dispose that are placed into test suite class.

### CODE REVIEW

File: src/Exercism.TestRunner.CSharp/TestsRewriter.cs [hunk 1]
```diff
@@ -204,7 +204,7 @@ public override SyntaxNode VisitClassDeclaration(ClassDeclarationSyntax node)
                                 IdentifierName("IDisposable"))));
 
                 // TODO: refactor to method
-                if (node.DescendantNodes().OfType<MethodDeclarationSyntax>().All(method => method.Identifier.Text != "Dispose"))
+                if (node.DescendantNodes().Where(n => n.Parent?.Equals(node) ?? false).OfType<MethodDeclarationSyntax>().All(method => method.Identifier.Text != "Dispose"))
```
Comment (ErikSchierboom) [inline src/Exercism.TestRunner.CSharp/TestsRewriter.cs:207]: Could you see if maybe the fix could be to use `ChildNodes()`? (see https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.syntaxnode.childnodes?view=roslyn-dotnet-3.11.0)

```suggestion
                if (node.ChildNodes().OfType<MethodDeclarationSyntax>().All(method => method.Identifier.Text != "Dispose"))
```

**Review by ErikSchierboom — COMMENTED** [review]
One tiny suggestion!

File: src/Exercism.TestRunner.CSharp/TestsRewriter.cs (hunk 1 above)
Comment (kgorbenko) [reply src/Exercism.TestRunner.CSharp/TestsRewriter.cs]: Yes, it seems to be the right solution. I was looking for something like this, but in the wrong place as it turns out.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
Great work!

**Pull request merged**